src/test/java/sixgaezzang/sidepeek/skill/serivce/SkillServiceTest.java

```diff
@@ -0,0 +1,93 @@
+import org.junit.jupiter.api.DisplayNameGeneration;
+import org.junit.jupiter.api.DisplayNameGenerator;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+import org.junit.jupiter.params.provider.NullAndEmptySource;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.transaction.annotation.Transactional;
+import sixgaezzang.sidepeek.skill.domain.Skill;
+import sixgaezzang.sidepeek.skill.dto.response.SkillResponse;
+import sixgaezzang.sidepeek.skill.repository.SkillRepository;
+
+@SpringBootTest
+@Transactional
+@DisplayNameGeneration(DisplayNameGenerator.ReplaceUnderscores.class)
+class SkillServiceTest {
+
+    @Autowired
+    SkillService skillService;
+    @Autowired
+    SkillRepository skillRepository;
+
+    @Nested
+    class 기술_스택_검색_테스트 {
+
+        static final int SKILL_COUNT = 5;
+        static final String[] skills = {"spring", "spring boot", "spring web", "react", "react query"};
+
+        @BeforeEach
+        void setUp() {
+            for (int i = 0; i < SKILL_COUNT; i++) {
+                Skill skill = createSkill(skills[i], "url");
+                skillRepository.save(skill);
+            }
+        }
+
+        @ParameterizedTest(name = "[{index}] {0}으로 검색할 때 " + SKILL_COUNT + "개의 모든 기술 스택이 나온다.")
+        @NullAndEmptySource
+        void 검색어_없이_전체_기술_스택_검색에_성공한다(String keyword) {
+            // given, when
+            List<SkillResponse> skills = skillService.searchByName(keyword)
+                .skills();
+
+            // then
+            assertThat(skills.size()).isEqualTo(SKILL_COUNT);
+        }
+
+        @ParameterizedTest(name = "[{index}] {0}으로 검색할 때 {1}개의 기술 스택이 나온다.")
+        @CsvSource(value = {"spring:3", "react:2"}, delimiter = ':')
+        void 검색어로_기술_스택_검색에_성공한다(String keyword, int count) {
+            // given, when
+            List<SkillResponse> skills = skillService.searchByName(keyword)
+                .skills();
+
+            // then
+            assertThat(skills.size()).isEqualTo(count);
+        }
+
+        @Test
+        void 검색어_최대_글자_수가_넘어_기술_스택_검색에_실패한다() {
+            // given
+            int keywordLength = MAX_SKILL_NAME_LENGTH + 1;
+            String keyword = "a".repeat(keywordLength);
+
+            // when
+            ThrowableAssert.ThrowingCallable search = () -> skillService.searchByName(keyword);
+
+            // then
+            assertThatExceptionOfType(IllegalArgumentException.class).isThrownBy(search)
+                .withMessage("최대 " + MAX_SKILL_NAME_LENGTH + "자의 키워드로 검색할 수 있습니다.");
+        }
+
+    }
+
+    private Skill createSkill(String name, String iconImageUrl) {
+        return Skill.builder()
+            .name(name)
+            .iconImageUrl(iconImageUrl)
+            .build();
+    }
```

Comment:
```suggestion
    private Skill createSkill(String name, String iconImageUrl) {
        Skill skill = Skill.builder()
            .name(name)
            .iconImageUrl(iconImageUrl)
            .build();
            
         return skillRepository.save(skill);  
    }
```

이렇게 하면 위 @BeforeEach 코드가 간결해질 것 같고, 재사용도 용이할 것 같아요!

Reply:
UserServiceTest와 통일성을 주고자 밖으로 빼놓긴 했습니다! 스킬 쪽은 사용자가 직접 저장하는  일은 없어서 같이 넣어줘도 될 것 같네용! 바로 반영할게유!!!

Reply:
리팩토링 데이때, 공통으로 사용되는 메서드에 대해 테스트 픽스처 분리를 하면 좋을 것 같아요!